---
title: "vbj-data-process"
author: "Genevieve Housman"
date: "April 2022"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Pseudobulk data for van Bakel and Jabs - Data Processing

Data include 10X 3' RNA-seq reads from 6 humans and 6 chimpanzees (raw data available at GSE181744):

* GHO-1:  H1-I + C1-I (iPSCs from H23555 and C8861)
* GHO-2:  H1-M + C1-M (iPSC-derived MSCs from H23555 and C8861)
* GHO-4:  H1-O + C1-O (iPSC-MSC-derived osteogenic cells from H23555 and C8861)
* HOU-10: H1-I-r2 + C1-I-r2 (iPSCs from H23555 and C8861)
* HOU-12: H1-M-r2 + C1-M-r2 (iPSC-derived MSCs from H23555 and C8861)
* HOU-16: H1-O-r2 + C1-O-r2 (iPSC-MSC-derived osteogenic cells from H23555 and C8861)
* HOU-1:  H2-I + C2-I (iPSCs from H20157 and C3647)
* HOU-2:  H2-M + C2-M (iPSC-derived MSCs from H20157 and C3647)
* HOU-4:  H2-O + C2-O (iPSC-MSC-derived osteogenic cells from H20157 and C3647)
* HOU-9:  H3-I + C3-I (iPSCs from H28126 and C3649)
* HOU-11: H3-M + C3-M (iPSC-derived MSCs from H28126 and C3649)
* HOU-15: H3-O + C3-O (iPSC-MSC-derived osteogenic cells from H28126 and C3649)
* HOU-5:  H4-I + C4-I (iPSCs from H28834 and C40210)
* HOU-6:  H4-M + C4-M (iPSC-derived MSCs from H28834 and C40210)
* HOU-8:  H4-O + C4-O (iPSC-MSC-derived osteogenic cells from H28834 and C40210)
* HOU-17:	H5-I + C5-I (iPSCs from H21792 and C40280)
* HOU-19:	H5-M + C5-M (iPSC-derived MSCs from H21792 and C40280)
* HOU-21:	H5-O + C5-O (iPSC-MSC-derived osteogenic cells from H21792 and C40280)
* HOU-18:	H6-I + C6-I (iPSCs from H20961 and C3624)
* HOU-20:	H6-M + C6-M (iPSC-derived MSCs from H20961 and C3624)
* HOU-22:	H6-O + C6-O (iPSC-MSC-derived osteogenic cells from H20961 and C3624)

For downstream analysis of 10X data, we need to:

* process fastq files into count matrices
* process count matrices for use in R

## Process fastq files into count matrices

Fastq files were processed into count matrices (genes x cells) using cellranger 3.1.0.

Cellranger reference was made using hg38 (all annotated genes in human genome):

```{bash, eval=FALSE, echo=TRUE}
wget ftp://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz -O hg38.fa.gz
gunzip hg38.fa.gz
wget ftp://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/refGene.txt.gz -O hg38.txt.gz
gunzip hg38.txt.gz
./genePredToGtf file hg38.input hg38.out
cat hg38.out | sort -k1,1 -k4,4n > hg38.gtf
rm hg38.txt
rm hg38.input
rm hg38.out
```

Count matrices were made using the hg38 cellranger reference:

Notes on what cellranger does to compute count matrices:

* genome alignment
	 + aligner (STAR) peforms splicing-aware alignment of reads to genome
	 + transcript annotation (GTF) buckets reads into exonic/intronic/intergenic and whether reads align (confidently) to genome
	 + [read = exonic if >=50% of it intersects an exon]
	 + [read = intronic if it is non-exonic and intersects an intron]
	 + [read = intergenic in all other cases]
* MAPQ adjustment (MAPQ 255)
   + for reads that align to a single exonic locus and to 1+ non-exonic loci, the exonic locus is prioritized and read is considered to be confidently mapped to exonic locus
* transcriptome alignment
	 + aligns exonic reads to annotated transcripts and looks for compatibility
	 + read is mapped to transcriptome if: read that is compatible with exons of annotated transcript and aligned to same strand
	 + read is uniquely (confidently) mapped to transcriptome if: read is compatible with a single gene annotation
	 + [uniquely (confidently) mapped reads are only ones considered for UMI counting]
* UMI counting
	 + before counting UMIs, corrections for sequencing errors in UMI sequences are attempted
	 + reads confidently mapped to transcriptome are grouped according to shared barcodes, UMIs, and gene annotations
	 + filter 1: if two groups of reads have same barcode and gene, but UMIs differ by a single base (Hamming distance 1 apart), then one UMI was likely introduced by substitution error in sequencing, and UMI of less-supported read group is corrected to UMI with higher support
	 + reads confidently mapped to transcriptome are grouped according to shared barcodes, UMIs (possibly corrected), and gene annotations
	 + filter 2: if two or more groups of reads have same barcode and UMI, but different gene annotations, gene annotation with most supporting reads is kept for UMI counting and other read groups are discarded (ties for maximal read support: all read groups are discarded as gene cannot be confidently assigned)
	 + each observed barcode-UMI-gene combination is recorded as UMI count in unfiltered feature-barcode matrix (number of reads supporting each counted UMI also recorded in molecule info file)
* calling cell barcodes
	 + [improved cell-calling algorithm that better identifies populations of low RNA content cells, especially when low RNA content cells are mixed into population of high RNA content cells, based on EmptyDrops method by Lun et al. 2018]
	 + step 1:
		  - uses cutoff based on total UMI counts of each barcode to identify cells - identifies primary mode of high RNA content cells
		  - (takes as input expected number of recovered cells, N)
		  - (let m be 99th percentile of top N barcodes by total UMI counts)
		  - (all barcodes whose total UMI counts exceed m/10 are called as cells)
	 + step 2:
		  - uses RNA profile of each remaining barcode to determine if it is "empty" or cell containing partition - captures low RNA content cells whose total UMI counts may be similar to empty GEMs
		  - (creates background model of RNA profiles from selected barcodes - multinomial distribution over genes - using Simple Good-Turing smoothing to provide a non-zero model estimate for genes not observed in representative empty GEM set)
		  - (RNA profile of each barcode not called as cell in step 1 is compared to background model)
		  - (barcodes whose RNA profile strongly disagrees with background model are added to set of positive cell calls)
		  - (to remedy cases in which set of barcodes called as cells does not match desired set of barcodes based on visual inspection: re-run count/reanalyze with --force-cells option, select desired barcodes from raw feature-barcode matrix in downstream analysis, or select custom barcodes in reanalyze using --barcodes)
* estimating multiplet rates
	 + cell-associated barcodes are classified as species1 or species2 based on which genome has more total UMI counts for that barcode
	 + cell-associated barcodes with total UMI counts that exceed the 10th percentile of the distributions for both species1 and species2 are classified as multiplets
	 + computes inferred multiplet rate by estimating the total number of multiplets (including (species1, species1) and (species2, species2))
		  - estimates via ML the total number of multiplet GEMs from the observed multiplets and the inferred ratio of species1 to species2 cells
		  - if ratio is 1:1, the inferred multiplet rate is approximately twice the observed (species1, species2) multiplets

```{bash, eval=FALSE, echo=TRUE}
cd /project2/gilad/ghousman/skeletal-human-chimp/scRNA/cellranger-data-vbj
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh GHO-1 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/190621_K00242_0576_AH7G3JBBXY_YG-GH-10X-4S-ln1,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190626_K00242_0578_AH77TNBBXY_YG-GH-10X-4S-ln2,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190726_K00242_0593_BH77N2BBXY_YG-GH-10X-4S-xtra-ln1,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190730_K00242_0594_BH7N3HBBXY_YG-GH-10X-4S-xtra-ln2
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh GHO-2 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/190621_K00242_0576_AH7G3JBBXY_YG-GH-10X-4S-ln1,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190626_K00242_0578_AH77TNBBXY_YG-GH-10X-4S-ln2,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190726_K00242_0593_BH77N2BBXY_YG-GH-10X-4S-xtra-ln1,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190730_K00242_0594_BH7N3HBBXY_YG-GH-10X-4S-xtra-ln2
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh GHO-3 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/190621_K00242_0576_AH7G3JBBXY_YG-GH-10X-4S-ln1,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190626_K00242_0578_AH77TNBBXY_YG-GH-10X-4S-ln2,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190726_K00242_0593_BH77N2BBXY_YG-GH-10X-4S-xtra-ln1,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190730_K00242_0594_BH7N3HBBXY_YG-GH-10X-4S-xtra-ln2
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh GHO-4 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/190621_K00242_0576_AH7G3JBBXY_YG-GH-10X-4S-ln1,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190626_K00242_0578_AH77TNBBXY_YG-GH-10X-4S-ln2,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190726_K00242_0593_BH77N2BBXY_YG-GH-10X-4S-xtra-ln1,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/190730_K00242_0594_BH7N3HBBXY_YG-GH-10X-4S-xtra-ln2
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-1 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-2 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-4 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-5 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-6 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-8 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-9 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-10 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-11 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-12 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-15 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-16 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-17 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191227_K00242_0655_BHG3MCBBXY-YG-GH-10X-HOU2-6s-lns12/Unaligned_YG-GH-10X-HOU2-6S/HG3MCBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200103_K00242_0656_BHFYVWBBXY_YG-GH-10X-HOU2-6S-ln3/Unaligned_YG-GH-10X-HOU2-6S-ln3/HFYVWBBXY_7,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200124_K00242_0661_BHGGF3BBXY-YG-GH-HOU2-10X-6S-ln4/Unaligned_YG-GH-10X-HOU2-6S-ln4/HGGF3BBXY_7
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-18 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191227_K00242_0655_BHG3MCBBXY-YG-GH-10X-HOU2-6s-lns12/Unaligned_YG-GH-10X-HOU2-6S/HG3MCBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200103_K00242_0656_BHFYVWBBXY_YG-GH-10X-HOU2-6S-ln3/Unaligned_YG-GH-10X-HOU2-6S-ln3/HFYVWBBXY_7,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200124_K00242_0661_BHGGF3BBXY-YG-GH-HOU2-10X-6S-ln4/Unaligned_YG-GH-10X-HOU2-6S-ln4/HGGF3BBXY_7
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-19 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191227_K00242_0655_BHG3MCBBXY-YG-GH-10X-HOU2-6s-lns12/Unaligned_YG-GH-10X-HOU2-6S/HG3MCBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200103_K00242_0656_BHFYVWBBXY_YG-GH-10X-HOU2-6S-ln3/Unaligned_YG-GH-10X-HOU2-6S-ln3/HFYVWBBXY_7,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200124_K00242_0661_BHGGF3BBXY-YG-GH-HOU2-10X-6S-ln4/Unaligned_YG-GH-10X-HOU2-6S-ln4/HGGF3BBXY_7
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-20 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191227_K00242_0655_BHG3MCBBXY-YG-GH-10X-HOU2-6s-lns12/Unaligned_YG-GH-10X-HOU2-6S/HG3MCBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200103_K00242_0656_BHFYVWBBXY_YG-GH-10X-HOU2-6S-ln3/Unaligned_YG-GH-10X-HOU2-6S-ln3/HFYVWBBXY_7,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200124_K00242_0661_BHGGF3BBXY-YG-GH-HOU2-10X-6S-ln4/Unaligned_YG-GH-10X-HOU2-6S-ln4/HGGF3BBXY_7
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-21 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191227_K00242_0655_BHG3MCBBXY-YG-GH-10X-HOU2-6s-lns12/Unaligned_YG-GH-10X-HOU2-6S/HG3MCBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200103_K00242_0656_BHFYVWBBXY_YG-GH-10X-HOU2-6S-ln3/Unaligned_YG-GH-10X-HOU2-6S-ln3/HFYVWBBXY_7,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200124_K00242_0661_BHGGF3BBXY-YG-GH-HOU2-10X-6S-ln4/Unaligned_YG-GH-10X-HOU2-6S-ln4/HGGF3BBXY_7
sbatch ./../../human-skeletal-scRNA/code/submit-cellranger-count.sh HOU-22 /project2/gilad/ghousman/references/hg38 /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191227_K00242_0655_BHG3MCBBXY-YG-GH-10X-HOU2-6s-lns12/Unaligned_YG-GH-10X-HOU2-6S/HG3MCBBXY_78,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200103_K00242_0656_BHFYVWBBXY_YG-GH-10X-HOU2-6S-ln3/Unaligned_YG-GH-10X-HOU2-6S-ln3/HFYVWBBXY_7,/project2/gilad/ghousman/skeletal-human-chimp/scRNA/200124_K00242_0661_BHGGF3BBXY-YG-GH-HOU2-10X-6S-ln4/Unaligned_YG-GH-10X-HOU2-6S-ln4/HGGF3BBXY_7

```

## Processing count matrices

Compile 10X data using seurat.

Each dataset being evaluated comprises gene count data that were calculated using the hg38 human genome.

In order to isolate just the human cells (by removing the chimpanzee cells), we used the species assignments determined in Housman et al (2022) PLOS Genetics (metadata available at GSE181744).

```{r, eval=FALSE, echo=TRUE}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Load batch info
batch <- read.csv(file='./data/scrna-batch.csv', header=TRUE, sep=",")
#batch <- as.data.frame(t(c("1-C","1","Chondrocyte","GHO-3","H1-C","H23555","M",22,"C1-C","C8861","M",16)))
#colnames(batch) <- c("Pooled_Sample_Name","Batch","Cell_Type","Sample_Name_at_Core",
#                     "Human_Sample_in_Pool","Human_Individual_in_Pool","Human_Sex_in_Pool","Human_Age_in_Pool",
#                     "Chimp_Sample_in_Pool","Chimp_Individual_in_Pool","Chimp_Sex_in_Pool","Chimp_Age_in_Pool")

#Isolate human and chimp cells for each collection
i <- 1
while (i <= length(batch$Sample_Name_at_Core)) {

  print(paste0("Processing ",batch$Sample_Name_at_Core[i]))
  
  #(0)define directory
  dir_data <- paste0('./../scRNA/cellranger-data-vbj/',batch$Sample_Name_at_Core[i],'/outs')
    
  #(1)load data (28278 features, 6794880 samples)
  data_raw <- Read10X(data=paste0(dir_data,"/raw_feature_bc_matrix"))
  
  #(2)fix barcodes
  colnames(data_raw) <- sapply(colnames(data_raw), gsub, pattern="-1", replacement="")
  
  #(3)load species assignment information (use predetermined species assignments from GSE181744)
  dir_spp <- paste0('./../scRNA/cellranger-data-full/cellranger-data-HumanOrthoV2_and_ChimpOrthoV2/',batch$Sample_Name_at_Core[i],'/outs/analysis/')
  spp <- read.csv(file=paste0(dir_spp,'gem_classification_GHedit.csv'))
  table(spp$NEWcall)
  
  #(4)subset data to only human cells
  table(colnames(data_raw) %in% spp$barcode[which(spp$NEWcall=="Human")])
  h.data_raw <- data_raw[,which(colnames(data_raw) %in% spp$barcode[which(spp$NEWcall=="Human")])]

  #(5)create seurat object
  data <- CreateSeuratObject(counts=h.data_raw)
    
  #(6)view details of seurat object (28278 features, X samples --> only human cells at this point out of the original 6794880 samples))
  data
    
  #(7)fix barcodes
  rownames(data@meta.data) <- sapply(rownames(data@meta.data), gsub, pattern="-1", replacement="")
  
  #(8)clear raw data from workspace
  rm(data_raw,h.data_raw)
  
  #(9)add species assignments to data
  m <- match(rownames(data@meta.data),spp$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  data <- AddMetaData(data, spp[m,]$NEWcall, col.name="Species")

  #(10)view details of species assignments
  table(data@meta.data$Species, useNA="always")

  #(11)save object
  saveRDS(data, file=paste0('./data/h.data.',batch$Sample_Name_at_Core[i],'.rds'))
  
  #(12)remove unnecessary variables
  rm(dir_data,data,dir_spp,spp)

  i <- i+1
  
}

```

### PICK UP FROM HERE
Add metadata to count matrices.

```{r process scRNA data - add mitochondrial metadata, eval=FALSE, echo=TRUE}

#Read in files
data <- list()
for (i in 1:length(batch$Sample_Name_at_Core)) {
  print(i)
  data[[i]] <- readRDS(paste0("./data/cellranger-data-full/h.c.dat.spp.",batch$Sample_Name_at_Core[i],".rds"))
}

#Identify mitochondrial genes (should be 12 genes total) and add percent of mitochondrial reads vs. total reads
#which(sapply(data[[i]]@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="MT-")==1)
#MT-CO2   16608
#MT-CYB   16616
#MT-ND2   16635
#MT-ND5   16643
#MT-CO1   16654
#MT-ND3   16676
#MT-ND4   16701
#MT-ND1   16703
#MT-ATP6  16709
#MT-CO3   16732
#MT-ND4L  17250
#MT-ATP8  17689
#INMT-MINDY4  18276 - not actually a mitochondrial gene
mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
for (i in 1:length(data)) { data[[i]]$percent.mt <- PercentageFeatureSet(data[[i]], features=mito) }

```

```{r load files with other relevant metadata, eval=FALSE, echo=TRUE}

#Load batch data
batch <- read.csv(file='./data/scrna-batch-reformat.csv', header=TRUE, sep=",")
batch[,c(1:7,9:10,21,24:25)] <- lapply(batch[,c(1:7,9:10,21,24:25)], function(x) as.character(x)) #names, sex, dates
batch[,c(8,11:20,22:23,25:26)] <- lapply(batch[,c(8,11:20,22:23,25:26)], function(x) as.numeric(as.character(x))) #numeric values only

```

```{r process scRNA data - add other relevant metadata, eval=FALSE, echo=TRUE}

#Add additional details to list

#collection
for (i in 1:length(data)) { data[[i]]$Collection <- i }

#cell type anticipated based on stage of differentiation at collection
for (i in c(1,4,7,10,13,16,19)) { data[[i]]$Cell.Type <- "iPSC" }
for (i in c(2,5,8,11,14,17,20)) { data[[i]]$Cell.Type <- "MSC" }
for (i in c(3,6,9,12,15,18,21)) { data[[i]]$Cell.Type <- "Osteogenic" }

#individual cell line used in collection
for (i in 1:length(data)) { data[[i]]$Individual <- NA }
for (i in 1:6) { data[[i]]$Individual[which(data[[i]]$Species=="Human")] <- "H1" }
for (i in 7:9) { data[[i]]$Individual[which(data[[i]]$Species=="Human")] <- "H2" }
for (i in 10:12) { data[[i]]$Individual[which(data[[i]]$Species=="Human")] <- "H3" }
for (i in 13:15) { data[[i]]$Individual[which(data[[i]]$Species=="Human")] <- "H4" }
for (i in 16:18) { data[[i]]$Individual[which(data[[i]]$Species=="Human")] <- "H5" }
for (i in 19:21) { data[[i]]$Individual[which(data[[i]]$Species=="Human")] <- "H6" }

#replicate in overall study design
for (i in c(1:3,7:21)) { data[[i]]$Replicate <- "1" }
for (i in 4:6) { data[[i]]$Replicate <- "2" }

#abbreviation of sample name in collection
for (i in 1:length(data)) { data[[i]]$Sample <- NA }
for (i in 1:length(data)) { 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="iPSC" & data[[i]]$Individual=="H1" & data[[i]]$Replicate=="1")] <- "H1-I" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="iPSC" & data[[i]]$Individual=="H1" & data[[i]]$Replicate=="2")] <- "H1-I-r2" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="iPSC" & data[[i]]$Individual=="H2" & data[[i]]$Replicate=="1")] <- "H2-I" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="iPSC" & data[[i]]$Individual=="H3" & data[[i]]$Replicate=="1")] <- "H3-I" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="iPSC" & data[[i]]$Individual=="H4" & data[[i]]$Replicate=="1")] <- "H4-I" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="iPSC" & data[[i]]$Individual=="H5" & data[[i]]$Replicate=="1")] <- "H5-I" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="iPSC" & data[[i]]$Individual=="H6" & data[[i]]$Replicate=="1")] <- "H6-I"
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="MSC" & data[[i]]$Individual=="H1" & data[[i]]$Replicate=="1")] <- "H1-M" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="MSC" & data[[i]]$Individual=="H1" & data[[i]]$Replicate=="2")] <- "H1-M-r2" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="MSC" & data[[i]]$Individual=="H2" & data[[i]]$Replicate=="1")] <- "H2-M" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="MSC" & data[[i]]$Individual=="H3" & data[[i]]$Replicate=="1")] <- "H3-M" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="MSC" & data[[i]]$Individual=="H4" & data[[i]]$Replicate=="1")] <- "H4-M" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="MSC" & data[[i]]$Individual=="H5" & data[[i]]$Replicate=="1")] <- "H5-M" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="MSC" & data[[i]]$Individual=="H6" & data[[i]]$Replicate=="1")] <- "H6-M"
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="Osteogenic" & data[[i]]$Individual=="H1" & data[[i]]$Replicate=="1")] <- "H1-O" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="Osteogenic" & data[[i]]$Individual=="H1" & data[[i]]$Replicate=="2")] <- "H1-O-r2" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="Osteogenic" & data[[i]]$Individual=="H2" & data[[i]]$Replicate=="1")] <- "H2-O" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="Osteogenic" & data[[i]]$Individual=="H3" & data[[i]]$Replicate=="1")] <- "H3-O" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="Osteogenic" & data[[i]]$Individual=="H4" & data[[i]]$Replicate=="1")] <- "H4-O" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="Osteogenic" & data[[i]]$Individual=="H5" & data[[i]]$Replicate=="1")] <- "H5-O" 
  data[[i]]$Sample[which(data[[i]]$Cell.Type=="Osteogenic" & data[[i]]$Individual=="H6" & data[[i]]$Replicate=="1")] <- "H6-O"
}

samples <- c("H1-I","H1-I-r2","H2-I","H3-I","H4-I","H5-I","H6-I",
             "H1-M","H1-M-r2","H2-M","H3-M","H4-M","H5-M","H6-M",
             "H1-O","H1-O-r2","H2-O","H3-O","H4-O","H5-O","H6-O")

#several details on cell line and collection
for (i in 1:length(data)) { data[[i]]$Sex <- NA }
for (i in 1:length(data)) { data[[i]]$Age <- NA }
for (i in 1:length(data)) { 
  for(j in samples) {
    if(j %in% batch$Sample_in_Pool) {
      data[[i]]$Sex[which(data[[i]]$Sample==j)] <- batch$Sex_in_Pool[which(batch$Sample_in_Pool==j)]
      data[[i]]$Age[which(data[[i]]$Sample==j)] <- batch$Age_in_Pool[which(batch$Sample_in_Pool==j)]
    }
  }
}

#cell classification information - stage of differentiation at collection
for (i in c(1,4,7,10,13,16,19)) { data[[i]]$Stage <- "Time 0" }
for (i in c(2,5,8,11,14,17,20)) { data[[i]]$Stage <- "Time 1" }
for (i in c(3,6,9,12,15,18,21)) { data[[i]]$Stage <- "Time 2" }

```

```{r, eval=FALSE, echo=TRUE}

#Save data
saveRDS(data, file="./data/cellranger-data-full/data.rds")

```

### Export processed count matrices for GEO

Processed data for pilot eQTL paper

```{r, eval=FALSE}

#Load and add metadata to GHO-3
dataGHO3 <- readRDS(paste0("./data/cellranger-data-full/h.data.GHO-3.rds"))
#percent mitochondria
mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
dataGHO3$percent.mt <- PercentageFeatureSet(dataGHO3, features=mito)
#collection
dataGHO3$Collection <- 0
#cell type anticipated based on stage of differentiation at collection
dataGHO3$Cell.Type <- "Chondrogenic"
#individual cell line used in collection
dataGHO3$Individual <- NA
dataGHO3$Individual[which(dataGHO3$Species=="Human")] <- "H1"
dataGHO3$Individual[which(dataGHO3$Species=="Chimp")] <- "C1"
#replicate in overall study design
dataGHO3$Replicate <- "1"
#msc media
dataGHO3$MSCmedia <- "3B"
#abbreviation of sample name in collection
dataGHO3$Sample <- NA
dataGHO3$Sample[which(dataGHO3$Individual=="H1")] <- "H1-C" 
dataGHO3$Sample[which(dataGHO3$Individual=="C1")] <- "C1-C" 
#several details on cell line and collection
dataGHO3$Sex <- "M"
dataGHO3$Age <- NA
dataGHO3$Age[which(dataGHO3$Individual=="H1")] <- 22
dataGHO3$Age[which(dataGHO3$Individual=="C1")] <- 16
dataGHO3$DateCollection <- "6/5/2019"
dataGHO3$Number6W <- 3
dataGHO3$PrewashCount <- NA
dataGHO3$PrewashCount[which(dataGHO3$Individual=="H1")] <- 1025000
dataGHO3$PrewashCount[which(dataGHO3$Individual=="C1")] <- 0.575
dataGHO3$PrewashViability <- NA
dataGHO3$PrewashViability[which(dataGHO3$Individual=="H1")] <- 2440000
dataGHO3$PrewashViability[which(dataGHO3$Individual=="C1")] <- 0.48
dataGHO3$PostwashCount <- NA
dataGHO3$PostwashCount[which(dataGHO3$Individual=="H1")] <- 440000
dataGHO3$PostwashCount[which(dataGHO3$Individual=="C1")] <- 445000
dataGHO3$PostwashViability <- NA
dataGHO3$PostwashViability[which(dataGHO3$Individual=="H1")] <- 0.75
dataGHO3$PostwashViability[which(dataGHO3$Individual=="C1")] <- 0.6
dataGHO3$PostmixCount <- 425000
dataGHO3$PostmixViability <- 0.75
dataGHO3$DatecDNA <- "6/5/2019"
dataGHO3$DateLibrary <- "6/10/2019"
dataGHO3$SequencingBatch <- 1
#facs information (control)
dataGHO3$CD90_Control <- NA
dataGHO3$CD73_Control <- NA
dataGHO3$CD105_Control <- NA
dataGHO3$CD45.CD34.CD11b.CD19.HLA.DR_Control <- NA
#facs information (cocktail) 
dataGHO3$CD90 <- NA
dataGHO3$CD73 <- NA
dataGHO3$CD105 <- NA
dataGHO3$CD45.CD34.CD11b.CD19.HLA.DR <- NA
#alizarin red data
dataGHO3$Alizarin <- NA
#oil red data (put with MSCs and osteogenic cells)
dataGHO3$OilRed <- NA
#cell classification information - stage of differentiation at collection
dataGHO3$Stage <- "Time 3"

#####

#Read in remaining data
data <- readRDS("./data/cellranger-data-full/data.rds")

#Rename cells
data[[1]] <- RenameCells(data[[1]], add.cell.id="GHO-1")
data[[2]] <- RenameCells(data[[2]], add.cell.id="GHO-2")
dataGHO3  <- RenameCells(dataGHO3, add.cell.id="GHO-3")
data[[3]] <- RenameCells(data[[3]], add.cell.id="GHO-4")

#Combine raw count data
tbl.cnt <- cbind(as.data.frame(GetAssayData(object=data[[1]], slot="counts")),
                 as.data.frame(GetAssayData(object=data[[2]], slot="counts")),
                 as.data.frame(GetAssayData(object=dataGHO3, slot="counts")),
                 as.data.frame(GetAssayData(object=data[[3]], slot="counts")))

#Combine metadata
tbl.met <- rbind(as.data.frame(data[[1]]@meta.data)[,-c(1:3)],
                 as.data.frame(data[[2]]@meta.data)[,-c(1:3)],
                 as.data.frame(dataGHO3@meta.data)[,-c(1:3)],
                 as.data.frame(data[[3]]@meta.data)[,-c(1:3)])

#Save processed data
write.table(tbl.cnt, 
            file="./data/cellranger-data-full/GHO_count.csv", 
            sep=',',
            row.names=T,
            col.names=T,
            quote=F)
write.table(tbl.met, 
            file="./data/cellranger-data-full/GHO_meta.csv", 
            sep=',',
            row.names=T,
            col.names=T,
            quote=F)

```

```{bash, eval=FALSE}
#easiest to do this on the computing cluster

#Make folder containing files to be uploaded to GEO
cd /project2/gilad/ghousman
mkdir geo_GHO_20210218
cd geo_GHO_20210218
mkdir data_raw
mkdir data_proc
cd data_proc
cp /project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/cellranger-data-full/GHO_count.csv ./GHO_count.csv
cp /project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/cellranger-data-full/GHO_meta.csv ./GHO_meta.csv
md5sum *
#manually added file checksum to metadata file
cd ..
cd data_raw
cp -r /project2/gilad/ghousman/skeletal-human-chimp/scRNA/190621_K00242_0576_AH7G3JBBXY_YG-GH-10X-4S-ln1/H7G3JBBXY_6/*/*.fastq.gz .
cp -r /project2/gilad/ghousman/skeletal-human-chimp/scRNA/190626_K00242_0578_AH77TNBBXY_YG-GH-10X-4S-ln2/H77TNBBXY_1/*/*.fastq.gz .
cp -r /project2/gilad/ghousman/skeletal-human-chimp/scRNA/190726_K00242_0593_BH77N2BBXY_YG-GH-10X-4S-xtra-ln1/H77N2BBXY_7/*/*.fastq.gz .
cp -r /project2/gilad/ghousman/skeletal-human-chimp/scRNA/190730_K00242_0594_BH7N3HBBXY_YG-GH-10X-4S-xtra-ln2/H7N3HBBXY_3/*/*.fastq.gz .
rm *_I1_*.fastq.gz
md5sum *
#manually added file checksum to metadata file
#manually transferred metadata file to geo_GHO_20210218 via FileZilla

#Upload files to GEO
cd /project2/gilad/ghousman
sftp geoftp@sftp-private.ncbi.nlm.nih.gov
cd uploads/ghousman_SbedAyUe
mkdir geo_GHO_20210218
put -R /project2/gilad/ghousman/geo_GHO_20210218

#Remove files for GEO
cd /project2/gilad/ghousman
rm -r geo_GHO_20210218
```

Processed data remaining for comparative osteogenesis paper

```{r, eval=FALSE}

#Export count matrix and metadata for remaining samples

#Load data
batch <- read.csv(file='./data/scrna-batch.csv', header=TRUE, sep=",")
data <- readRDS("./data/cellranger-data-full/data.rds")

#Rename cells and combine raw count data and metadata
tbl.cnt <- data.frame(matrix(NA, nrow=19377, ncol=0))
tbl.met <- data.frame(matrix(NA, nrow=0, ncol=33))
i=4
while (i <= length(data)) {
  print(i)
  data[[i]] <- RenameCells(data[[i]], add.cell.id=batch$Sample_Name_at_Core[i])
  tbl.cnt <- cbind(tbl.cnt,as.data.frame(GetAssayData(object=data[[i]], slot="counts")))
  tbl.met <- rbind(tbl.met,as.data.frame(data[[i]]@meta.data)[,-c(1:3)])
  i=i+1
}

#Save processed data
write.table(tbl.cnt, 
            file="./data/cellranger-data-full/HOU_count.csv", 
            sep=',',
            row.names=T,
            col.names=T,
            quote=F)
write.table(tbl.met, 
            file="./data/cellranger-data-full/HOU_meta.csv", 
            sep=',',
            row.names=T,
            col.names=T,
            quote=F)

```

```{bash, eval=FALSE}
#easiest to do this on the computing cluster
cd /project2/gilad/ghousman

#Make folder containing files to be uploaded to GEO
mkdir geo_GHO_20210807
cd geo_GHO_20210807
mkdir data_raw
mkdir data_proc
cd data_proc
cp /project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/cellranger-data-full/HOU_count.csv .
cp /project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/cellranger-data-full/HOU_meta.csv .
md5sum *
#manually added file checksum to metadata file
cd ..
cd data_raw
cp -r /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191112_K00242_0637_AHCLNGBBXY-YG-GH-10X-12S-HOU-lns1-2/HCLNGBBXY_78/*/*.fastq.gz .
cp -r /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123/*/*.fastq.gz .
#these files have the same names as some in the next folder [see notes below] so rename the files
for file in *.fastq.gz; do mv "$file" "${file//_L001_/_L005_}"; done
for file in *.fastq.gz; do mv "$file" "${file//_L002_/_L006_}"; done
for file in *.fastq.gz; do mv "$file" "${file//_L003_/_L009_}"; done
cp -r /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234/*/*.fastq.gz .
cp -r /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191227_K00242_0655_BHG3MCBBXY-YG-GH-10X-HOU2-6s-lns12/Unaligned_YG-GH-10X-HOU2-6S/HG3MCBBXY_78/*/*.fastq.gz .
#these files have the same names as some in the next two folders [see notes below] so rename the files
for file in *.fastq.gz; do mv "$file" "${file//HOU-17_S1_L007_/HOU-17_S1_L005_}"; done
for file in *.fastq.gz; do mv "$file" "${file//HOU-18_S2_L007_/HOU-18_S2_L005_}"; done
for file in *.fastq.gz; do mv "$file" "${file//HOU-19_S3_L007_/HOU-19_S3_L005_}"; done
for file in *.fastq.gz; do mv "$file" "${file//HOU-20_S4_L007_/HOU-20_S4_L005_}"; done
for file in *.fastq.gz; do mv "$file" "${file//HOU-21_S5_L007_/HOU-21_S5_L005_}"; done
for file in *.fastq.gz; do mv "$file" "${file//HOU-22_S6_L007_/HOU-22_S6_L005_}"; done
cp -r /project2/gilad/ghousman/skeletal-human-chimp/scRNA/200103_K00242_0656_BHFYVWBBXY_YG-GH-10X-HOU2-6S-ln3/Unaligned_YG-GH-10X-HOU2-6S-ln3/HFYVWBBXY_7/*/*.fastq.gz .
#these files have the same names as some in the next folder [see notes below] so rename the files
for file in *.fastq.gz; do mv "$file" "${file//HOU-17_S1_L007_/HOU-17_S1_L006_}"; done
for file in *.fastq.gz; do mv "$file" "${file//HOU-18_S2_L007_/HOU-18_S2_L006_}"; done
for file in *.fastq.gz; do mv "$file" "${file//HOU-19_S3_L007_/HOU-19_S3_L006_}"; done
for file in *.fastq.gz; do mv "$file" "${file//HOU-20_S4_L007_/HOU-20_S4_L006_}"; done
for file in *.fastq.gz; do mv "$file" "${file//HOU-21_S5_L007_/HOU-21_S5_L006_}"; done
for file in *.fastq.gz; do mv "$file" "${file//HOU-22_S6_L007_/HOU-22_S6_L006_}"; done
cp -r /project2/gilad/ghousman/skeletal-human-chimp/scRNA/200124_K00242_0661_BHGGF3BBXY-YG-GH-HOU2-10X-6S-ln4/Unaligned_YG-GH-10X-HOU2-6S-ln4/HGGF3BBXY_7/*/*.fastq.gz .
rm *_I1_*.fastq.gz
md5sum *
#manually added file checksum to metadata file
#manually transferred metadata file to geo_GHO_20210807 via FileZilla

#Upload files to GEO
cd /project2/gilad/ghousman
sftp geoftp@sftp-private.ncbi.nlm.nih.gov
cd uploads/ghousman_SbedAyUe
mkdir geo_GHO_20210807
put -R /project2/gilad/ghousman/geo_GHO_20210807

#Remove files for GEO
cd /project2/gilad/ghousman
rm -r geo_GHO_20210807

### FILE RENAMING NOTES
### NEED TO RENAME FILES TO L005/L006/L009 ###
### /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123/*/*_L001_*.fastq.gz
### /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123/*/*_L002_*.fastq.gz
### /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191217_K00242_0648_BHFHY7BBXY-YG-GH-10X-12S-HOU-lns345/FastQ/HFHY7BBXY_123/*/*_L003_*.fastq.gz
### vs.
### /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234/*/*_L001_*.fastq.gz
### /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234/*/*_L002_*.fastq.gz
### /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191220_K00242_0650_BHFTJ2BBXY-YG-GH-10X-12S-HOU-lns6789/FastQ/HFTJ2BBXY_1234/*/*_L003_*.fastq.gz
### NEED TO RENAME FILES TO L005/L006 ###
### /project2/gilad/ghousman/skeletal-human-chimp/scRNA/191227_K00242_0655_BHG3MCBBXY-YG-GH-10X-HOU2-6s-lns12/Unaligned_YG-GH-10X-HOU2-6S/HG3MCBBXY_78/*/*_L007_*.fastq.gz
### vs.
### /project2/gilad/ghousman/skeletal-human-chimp/scRNA/200103_K00242_0656_BHFYVWBBXY_YG-GH-10X-HOU2-6S-ln3/Unaligned_YG-GH-10X-HOU2-6S-ln3/HFYVWBBXY_7/*/*_L007_*.fastq.gz
### vs.
### /project2/gilad/ghousman/skeletal-human-chimp/scRNA/200124_K00242_0661_BHGGF3BBXY-YG-GH-HOU2-10X-6S-ln4/Unaligned_YG-GH-10X-HOU2-6S-ln4/HGGF3BBXY_7/*/*_L007_*.fastq.gz
```
